### ISSUE TITLE
Migrate to OpenLibrary as Metadata Backend

### ISSUE THREAD

**Issue by stephondoestech** [Readarr/Readarr#3486]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Is your feature request related to a problem? Please describe

Bookinfo is not stable and Goodreads is planning to revoke their [API](https://help.goodreads.com/s/article/Does-Goodreads-support-the-use-of-APIs) access with no clear shutoff date. 

### Describe the solution you'd like

The desired outcome would be to add OpenLibrary as a Metadata source and have Readarr leverage that as the backend for book metadata. 

### Describe alternatives you've considered

I've looked into [Hardcover](https://hardcover.app/) which looks promising, but didn't see enough support/adoption for it yet to recommend it here. 

### Anything else?

I have started working on this already leveraging [this](https://github.com/Luca3317/OpenLibrary.NET) library and so far have seen promising results.

**Comment by imi-kitten** [comment]
has there been any progress on this?

**Comment by bakerboy448** [comment]
https://wiki.servarr.com/readarr/metadata-issues

**Comment by ProfessorDey** [comment]
I do have to ask, having spent 3 days just rescanning and fighting with one small library, why in particular is this such a challenge? Especially when it comes to audiobook sources? Audiobookshelf has like 7-8 different metadata sources it can pull from and manages to do so almost instantly, so I'm genuinely curious what the technical limitation there is in terms of scraping basic metadata and why the system doesn't allow us to substitute our own.

I was hoping to use readarr along with Prowlarr to auto-update my audiobook library as new books come out, but the fact is it seems utterly unusable because it can't identify half of the books that came out in the last few months and doesn't recognise that audiobook versions even exist in most cases, while the metadata issues mean I can neither manually import nor specify specific searches for any given book/series/author.

Since it seems there's no problem on prowlarr's end in terms of pulling a complete list of everything a source has from a given author,, and audiobookshelf can immediately pull metadata from multiple sources easily, I'm genuinely confused why this seems to be so complicated on Readarr's end?

Literally all I want out of the software is the ability to directly say 'Download anything from x author on this source and name the file according to these rules unless that folder already exists in which case skip it,'

**Comment by Ejmathewp** [comment]
> Literally all I want out of the software is the ability to directly say 'Download anything from x author on this source and name the file according to these rules unless that folder already exists in which case skip it,'

The metadata has been completely unreliable for a long time, making Readarr nearly worthless. So much fun manually creating directories, renaming, and moving files when I have a service that is _supposed_ to do exactly that. Audiobookshelf picks things up flawlessly and pulls all the metadata, so clearly metadata providers aren't the issue.

**Comment by bakerboy448** [comment]
> there is in terms of scraping basic metadata

if it's so basic, then looking forward for you joining discord to quickly implement an entire new metadata server - migrating to OpenLibrary - and migrating existing libraries.

TL;DR it's not simple and among other reasons, scraping is largely forbidden

the previously linked wiki page and discord pins contain all the information available.

- given the noise/spam on this GHI and the related GHI
- given users' repeated inability to read prior updates
- given the unproductive, hostile comments

this is now locked.

all information is shared on the previously posted wiki page and in the discord pins

Re this issue and the migration to OpenLibrary

> If you wish to assist with this - please visit us on Discord, acquire the Tester role and meet us in the appropiate channel. Please do not ask for updates as that is not productive and updates will be shared as progress progresses.
